ext/mysql2/extconf.rb

```diff
@@ -81,8 +91,10 @@ def asplode(lib)
 
 if have_header('mysql.h')
   prefix = nil
+  add_ssl_defines('mysql.h')
 elsif have_header('mysql/mysql.h')
   prefix = 'mysql'
+  add_ssl_defines('mysql/mysql.h')
```

Comment:
Using pattern for `errmsg.h`, this could be written as:

```
add_ssl_defines([prefix, 'mysql.h'].compact.join '/')
```

Upside: only one line of code, and not inside a conditional block (move it below the if-have-header block). Downside: looks complicated.